test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,355 @@ package provisioner
+	if err != nil {
+		fmt.Println("Failed to create resource group:", err)
+		return err
+	}
+
+	AzureProps.ResourceGroup = &resourceGroup
+
+	log.Infof("Successfully Created Resource group %s.\n", AzureProps.ResourceGroupName)
+	return nil
+}
+
+func deleteResourceGroup() error {
+	log.Infof("Deleting Resource group %s.\n", AzureProps.ResourceGroupName)
+	_, err := AzureProps.ResourceGroupClient.Delete(context.Background(), AzureProps.ResourceGroupName)
+	if err != nil {
+		if isNotFoundError(err) {
+			return nil
+		}
+
+		fmt.Println("Failed to delete resource group:", err)
+		return err
+	}
+
+	log.Infof("Successfully Deleted Resource group %s.\n", AzureProps.ResourceGroupName)
+
+	return nil
+}
+
+func createVnetSubnet() error {
+	addressPrefix := "10.2.0.0/16"
+	subnetAddressPrefix := "10.2.0.0/24"
+	vnetParams := network.VirtualNetwork{
+		Location: &AzureProps.Location,
+		Name:     &AzureProps.VnetName,
+		VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
+			AddressSpace: &network.AddressSpace{
+				AddressPrefixes: &[]string{addressPrefix},
+			},
+			Subnets: &[]network.Subnet{
+				{
+					Name: &AzureProps.SubnetName,
+					SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+						AddressPrefix: &subnetAddressPrefix,
+					},
+				},
+			},
+		},
+	}
+
+	// Create the virtual network
+	log.Infof("Creating  vnet %s in resource group %s with addressPrefix %s subnetAddressPrefix %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName, addressPrefix, subnetAddressPrefix)
+	_, err := AzureProps.ManagedvnetClient.CreateOrUpdate(context.Background(), AzureProps.ResourceGroupName, AzureProps.VnetName, vnetParams)
+	if err != nil {
+		return err
+	}
+
+	subnet, err := AzureProps.ManagedsubnetClient.Get(context.Background(), AzureProps.ResourceGroupName, AzureProps.VnetName, AzureProps.SubnetName, "")
+	if err != nil {
+		return err
+	}
+
+	AzureProps.SubnetID = *subnet.ID
+
+	if len(AzureProps.SubnetID) <= 0 {
+		return errors.New("SubnetID is empty, unknown error happened when create Subnet.")
+	}
+
+	log.Infof("Successfully Created vnet %s in resource group %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName)
+
+	return nil
+}
+
+func deleteVNet() error {
+	log.Infof("Deleting vnet %s in resource group %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName)
+	_, err := AzureProps.ManagedvnetClient.Delete(context.Background(), AzureProps.ResourceGroupName, AzureProps.VnetName)
+	if err != nil {
+		if isNotFoundError(err) {
+			return nil
+		}
+		fmt.Println("Failed to delete the virtual network: ", err)
+		return err
+	}
+
+	log.Infof("Successfully Deleted vnet %s in resource group %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName)
+	return nil
+}
+
+func createResourceImpl() error {
+	err := createResourceGroup()
+	if err != nil {
+		return err
+	}
+
+	log.Info("waiting for the Resource group to be available before creating vnet...")
+	time.Sleep(60 * time.Second)
```

Comment:
can we have a constant instead of hardcoded value? And a comment explaining why 60 seconds were chosen?

Reply:
azure group creation take few seconds, but it worst cases it can go to 30 min, but we are creating simple resource group without any dependencies so it should get wrapped up in 1 minute and if not later step will fail saying it doesnt exist which is fine. 
Other thing we can do is use the context and add a piece of code to wait until rg with that ctx is created

Reply:
I think Suraj meant to add this to the code.